### PR TITLE
Mark LoadFrom overload that always throws as obsolete

### LINKED ISSUE #96410: Assembly.LoadFrom overload unconditionally throws NotSupportedException
### Description

The following API:
`public static Assembly LoadFrom(string assemblyFile, byte[]? hashValue, AssemblyHashAlgorithm hashAlgorithm)`
Unconditionally throws `NotSupportedException`.
This is a bad development experience. It looks like a valid API until it is used and it throws at runtime. Marking it as obsolete (or implementing it) would give the necessary development time signal to not use it rather than waste a developer's time.

### Reproduction Steps

`System.Reflection.Assembly.LoadFrom("", null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None);`

### Expected behavior

The call should either be functional or the API should be marked obsolete.

### Actual behavior

Code compiles with no warnings only to unconditionally throw an exception at runtime.

### Regression?

Not tested but this API presumably worked on .NET Framework.

### Known Workarounds

Don't use this API.

### Configuration

- .NET 8
- Windows 10
- x64
- Not specific to this configuration
- Not using Blazor

### Other information

_No response_

### PR DESCRIPTION
**Pull request by agocke**
Fixes #96410

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml [hunk 1]
```diff
@@ -1,6 +1,22 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Suppression>
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:4]: Unintentional change?

Comment (agocke) [reply src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:4]: Ugh, yes, forgot to review the generated suppressions.

**Review by jkotas — APPROVED** [review]

**Comment by huoyaoyuan** [comment]
Does this require change in reference source?

**Review by jkotas — CHANGES_REQUESTED** [review]
> Does this require change in reference source?

Good catch!

File: src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml [hunk 2]
```diff
@@ -29,4 +29,10 @@
     <DiagnosticId>CP0015</DiagnosticId>
     <Target>M:System.Diagnostics.Tracing.EventSource.WriteEventWithRelatedActivityIdCore(System.Int32,System.Guid*,System.Int32,System.Diagnostics.Tracing.EventSource.EventData*):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
   </Suppression>
+  <Suppression>
+    <DiagnosticId>CP0016</DiagnosticId>
+    <Target>M:System.Reflection.Assembly.LoadFrom(System.String,System.Byte[],System.Configuration.Assemblies.AssemblyHashAlgorithm):[T:System.ObsoleteAttribute]</Target>
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:34]: Can these CompatibilitySuppressions be listed just once in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/CompatibilitySuppressions.xml since they are identical for all CoreLib variants?

Comment (MichalStrehovsky) [reply src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:34]: Aren't these suppressions only needed because there's a difference between ref and def assembly? I would think after updating the ref assembly, these won't be needed anymore. I'm a bit suspicious about the other suppressions in this file too.

Comment (jkotas) [reply src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:34]: IIRC, these suppressions are sometimes needed to make the build happy about vPrevious -> vNext deltas. I am not sure whether it is the case here. I agree that the existing suppressions in this file look suspicious and they should not be needed.

Comment (agocke) [reply src/coreclr/System.Private.CoreLib/CompatibilitySuppressions.xml:34]: I can certainly try removing the suppression now that I've updated the ref assembly.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/CompatibilitySuppressions.xml [hunk 3]
```diff
@@ -865,4 +865,4 @@
     <DiagnosticId>CP0015</DiagnosticId>
     <Target>M:System.Diagnostics.Tracing.EventSource.WriteEventWithRelatedActivityIdCore(System.Int32,System.Guid*,System.Int32,System.Diagnostics.Tracing.EventSource.EventData*):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
   </Suppression>
-</Suppressions>
+</Suppressions>
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/CompatibilitySuppressions.xml:868]:
```suggestion
</Suppressions>
```
Unnecessary change

File: src/mono/System.Private.CoreLib/CompatibilitySuppressions.xml [hunk 4]
```diff
@@ -32,4 +32,4 @@
     <DiagnosticId>CP0015</DiagnosticId>
     <Target>M:System.Diagnostics.Tracing.EventSource.WriteEventWithRelatedActivityIdCore(System.Int32,System.Guid*,System.Int32,System.Diagnostics.Tracing.EventSource.EventData*):[T:System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute]</Target>
   </Suppression>
-</Suppressions>
+</Suppressions>
```
Comment (jkotas) [inline src/mono/System.Private.CoreLib/CompatibilitySuppressions.xml:35]:
```suggestion
</Suppressions>

```

**Review by jkotas — APPROVED** [review]

**Comment by vcsjones** [comment]
@agocke doesn't this need a breaking change doc so the SYSLIB0056 gets documented? I don't see it in https://github.com/dotnet/docs/blob/main/docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md

**Comment by agocke** [comment]
Good catch -- I will add that before we ship
